scripts/bytecodecompare/storebytecode.sh

```diff
@@ -28,18 +28,14 @@
 set -e
 
 REPO_ROOT="$(dirname "$0")"/../..
+cd "$REPO_ROOT"
```

Comment:
If you change the working directory, then please do it in a sub-shell. Running a script should not change the working directory.

Reply:
The whole script runs in one big subshell :) It does not change your working directory if you run it as `./storebytecode.sh`.

Unless you mean sourcing it? But in that case you should expect a lot more side-effects to persist anyway unless the script is specifically designed to be sourced.

Reply:
But maybe you just mean that it's a good practice to do `cd` within a subshell in a script so that the change does not persist when you move on to other parts of the script? I'd agree in general but it's a bit of a problem in this specific case:
- Variables assigned to in a subshell don't persist so `REPO_ROOT=$(pwd)` would have no effect.
- I could use `realpath` to get absolute path and get rid of `cd`. I considered it but I think `cd` was used here as a quick and dirty workaround for the usual problem - the clean solution does not work on macOS :)
- I could use a big subshell that would include the `TMPDIR` part but I think that would miss the point (i.e. the directory change would be visible there and I think this is want you wanted to avoid).

Reply:
What I want to avoid is that running this script from another script or the commandline results in the cwd being different. Why not put almost the whole script in a subshell?

Reply:
Because I don't think it's necessary and it would be weird to have to put every script in a subshell. But maybe I'm just missing something. I'm thinking of this situation:

```bash
echo "cd /tmp/" > test.sh
chmod +x test.sh

pwd
./test.sh
pwd
```

For me this prints:
```
/home/cameel
/home/cameel
```
so clearly the working directory was not changed. And `storebytecode.sh` after my change behaves the same way. Why do you think it would change the cwd of your shell session or the calling script?

Reply:
oh weird - you are absolutely right!